src/overlays/actors/ovl_En_Part/z_en_part.h

```diff
@@ -6,13 +6,15 @@
 
 struct EnPart;
 
+typedef void (*EnPartActionFunc)(struct EnPart*, GlobalContext*);
+
 typedef struct EnPart {
     /* 0x000 */ Actor actor;
-    /* 0x14C */ u8 unk_14C;
-    /* 0x14E */ s16 unk_14E;
-    /* 0x150 */ s32 unk_150;
-    /* 0x154 */ f32 unk_154;
-    /* 0x158 */ f32 unk_158;
+    /* 0x14C */ u8 action;
+    /* 0x14E */ s16 delay;
```

Comment:
I think this actor uses `delay` like "when `delay` reaches 0, stop doing things"
I'm not sure I like "delay" because that often implies "wait to do something" rather than "do something until"

For one of the actions (4, `func_80ACE7E8`) "stop doing things" doesn't mean "remove the actor" like for the others (it still draws), so `lifetime` isn't a good name
I'm not very inspired at the moment

Reply:
From my tests, delay was used as "wait until spawning effect". See also `func_80ACE13C` where all the function calls are effects.

Reply:
But doesn't `EnPart_Draw` get called meanwhile? It draws `this->displayList`

Reply:
I tested and it seems, at least for stalchild, that `EnPart_Draw` draws the bones. Since `delay` only matter for the flames, a good name might be `effectDelay`.